test/extended/util/docker.go

```diff
@@ -0,0 +1,68 @@
+package util
+
+import (
+	dockerClient "github.com/fsouza/go-dockerclient"
+	tutil "github.com/openshift/origin/test/util"
+	"errors"
+)
+
+func TagImage(tagee, tagor string) error {
+	client, dcerr := tutil.NewDockerClient()
+	if dcerr != nil {
+		return dcerr
+	}
+	opts := dockerClient.TagImageOptions{
+		Repo:  tagee,
+		Tag:   "latest",
+		Force: true,
+	}
+	terr := client.TagImage(tagor, opts)
+	return terr
+}
+
+func PullImage(name string) error {
+	client, dcerr := tutil.NewDockerClient()
+	if dcerr != nil {
+		return dcerr
+	}
+	opts := dockerClient.PullImageOptions{
+		Repository: name,
+		Tag:        "latest",
+	}
+	perr := client.PullImage(opts, dockerClient.AuthConfiguration{})
+	return perr
+}
+
+func GetImageIDForTags(comps []string) ([]string, error) {
+	client, dcerr := tutil.NewDockerClient()
+	if dcerr != nil {
+		return nil, dcerr
+	}
+	imageList, serr := client.ListImages(dockerClient.ListImagesOptions{})
+	if serr != nil {
+		return nil, serr
+	}
+
+	returnTags := make([]string, 0)
+	for _, comp :=  range comps {
+		var found bool
+		for _, image := range imageList {
+			for _, repTag := range image.RepoTags {
+				if repTag == comp {
+					found = true
+					returnTags = append(returnTags, image.ID)
```

Comment:
(that way use cases that need to fail immediately if any of them aren't found can just check the error without iterating the list, and use cases that can progress (or even expect some empty results) can continue)